gbn/timeout_manager_test.go

```diff
@@ -0,0 +1,369 @@
+package gbn
+
+import (
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/lightningnetwork/lnd/lntest/wait"
+	"github.com/stretchr/testify/require"
+)
+
+// BenchmarkTimeoutMgrSynchronously benchmarks the timeout manager when sending
+// and receiving messages synchronously.
+func BenchmarkTimeoutMgrSynchronously(b *testing.B) {
+	// Create a new timeout manager to use for the test. We set the timeout
+	// update frequency 2, so that the resend timeout is dynamically set
+	// every other message.
+	tm := NewTimeOutManager(nil, WithTimeoutUpdateFrequency(2))
+
+	for n := 0; n < b.N; n++ {
+		msg := &PacketData{Seq: uint8(n)}
+
+		tm.Sent(msg, false)
+		tm.Received(msg)
+	}
+}
+
+// BenchmarkTimeoutMgrConcurrently benchmarks the timeout manager when sending
+// and receiving messages concurrently.
+func BenchmarkTimeoutMgrConcurrently(b *testing.B) {
```

Comment:
nice! 

ok last last thing I promise 😂  can we add a TestStressTestTimoutMgr test that basically does this exact test with  very high N value. This will just make sure that this stress test is part of our CI and then if we ever actually do have a deadlock situation I dont think bench mark tests are currently run by the CI but I could be wrong), it would likely get caught by the test. it can literally just be:

```
func TestStressTestTimeoutMgr(t *testing.T) {
	tm := NewTimeOutManager(nil, WithTimeoutUpdateFrequency(2))

	var wg sync.WaitGroup
	for n := 0; n < 100000; n++ {
		wg.Add(1)
		go func(seq uint8) {
			defer wg.Done()

			msg := &PacketData{Seq: seq}

			tm.Sent(msg, false)
			tm.Received(msg)
		}(uint8(n))
	}

	wg.Wait()
}
```

Reply:
Ah good idea! Added the test with a slight modification to ensure it runs in parallel with the other tests :)!